src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -0,0 +1,142 @@
+        if (opprettOppgaverForBarn.isEmpty()) return
+        val gjeldendeBarnList =
+            gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(opprettOppgaverForBarn.map { it.behandlingId }).toSet()
+        gjeldendeBarnList.forEach { gjeldendeBarn ->
+            val opprettOppgaveForEksternId =
+                opprettOppgaverForBarn.firstOrNull { it.fødselsnummer == gjeldendeBarn.barnPersonIdent }
+            val finnesOppgave = oppgaveRepository.findByBehandlingIdAndBarnPersonIdentAndAlder(
+                gjeldendeBarn.behandlingId,
+                gjeldendeBarn.barnPersonIdent,
+                opprettOppgaveForEksternId?.alder
+            ) != null
+            if (!finnesOppgave && opprettOppgaveForEksternId != null) {
+                val opprettOppgaveRequest = lagOppgaveRequestForOppfølgingAvBarnFyltÅr(opprettOppgaveForEksternId, gjeldendeBarn)
+                val opprettetOppgaveId = oppgaveClient.opprettOppgave(opprettOppgaveRequest)
+                oppgaveClient.leggOppgaveIMappe(opprettetOppgaveId)
+                val oppgave = Oppgave(
+                    gsakOppgaveId = opprettetOppgaveId,
+                    behandlingId = gjeldendeBarn.behandlingId,
+                    barnPersonIdent = gjeldendeBarn.barnPersonIdent,
+                    type = Oppgavetype.InnhentDokumentasjon,
+                    alder = opprettOppgaveForEksternId.alder
+                )
+                oppgaveRepository.insert(oppgave)
+            }
+        }
+    }
+
+    private fun lagOppgaveRequestForOppfølgingAvBarnFyltÅr(
+        opprettOppgaveForEksternId: OpprettOppgaveForBarn,
+        barnTilOppgave: BarnTilOppgave
+    ) =
+        OpprettOppgaveRequest(
+            ident = OppgaveIdentV2(
+                ident = opprettOppgaveForEksternId.fødselsnummer,
+                gruppe = IdentGruppe.FOLKEREGISTERIDENT
+            ),
+            saksId = barnTilOppgave.eksternFagsakId.toString(),
+            tema = Tema.ENF,
+            oppgavetype = Oppgavetype.InnhentDokumentasjon,
+            fristFerdigstillelse = oppgaveService.lagFristForOppgave(LocalDateTime.now()),
+            beskrivelse = opprettOppgaveForEksternId.alder.oppgavebeskrivelse,
+            enhetsnummer = personopplysningerIntegrasjonerClient.hentNavEnhetForPersonMedRelasjoner(
+                opprettOppgaveForEksternId.fødselsnummerSøker
+            ).first().enhetId,
+            behandlingstema = opprettBehandlingstema(StønadType.OVERGANGSSTØNAD).value,
+            tilordnetRessurs = null,
+            behandlesAvApplikasjon = "familie-ef-sak"
+        )
+
+    fun opprettBehandlingstema(stønadstype: StønadType): Behandlingstema {
+        return Behandlingstema
+            .fromValue(
+                stønadstype.name.lowercase(Locale.getDefault())
+                    .replaceFirstChar { it.uppercase() }
+            )
+    }
+
+    private fun fødselsdato(barnTilUtplukkForOppgave: BarnTilUtplukkForOppgave): LocalDate? {
```

Comment:
Her må vi kanskje håndtere terminbarn også?

Reply:
Ja, tenker at det kan være lurt å hente ut fnr fra pdl for disse, da de burde ha fått fødselsnummer innen 6 mnd etter fødsel etter denne diskusjonen: https://nav-it.slack.com/archives/C01087QRJ2U/p1661778825749399